src/client.js

```diff
@@ -403,14 +403,29 @@ Client.prototype.inputLine = function(data) {
 };
 
 Client.prototype.more = function(data) {
-	var client = this;
-	var target = client.find(data.target);
+	const client = this;
+	const target = client.find(data.target);
+
 	if (!target) {
 		return;
 	}
-	var chan = target.chan;
-	var index = chan.messages.findIndex((val) => val.id === data.lastId);
-	var messages = chan.messages.slice(Math.max(0, index - 100), index);
+
+	const chan = target.chan;
+	const index = chan.messages.findIndex((val) => val.id === data.lastId);
+
+	// If we don't find the requested id, or it's first message we have
+	// Send an empty message array
+	if (index < 1) {
+		client.emit("more", {
+			chan: chan.id,
+			messages: []
+		});
+
+		return;
+	}
+
+	const messages = chan.messages.slice(Math.max(0, index - 100), index);
```

Comment:
Could we simplify this code by just doing

`const messages = index < 1 ? [] : chan.messages.slice(Math.max(0, index - 100), index);`

?